lib/filterx/filterx-pipe.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2023 Balázs Scheidler <balazs.scheidler@axoflow.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "filterx/filterx-pipe.h"
+#include "filterx/filterx-eval.h"
+#include "stats/stats-registry.h"
+
+typedef struct _LogFilterXPipe
+{
+  LogPipe super;
+  gchar *name;
+  GList *stmts;
+} LogFilterXPipe;
+
+static gboolean
+log_filterx_pipe_init(LogPipe *s)
+{
+  LogFilterXPipe *self = (LogFilterXPipe *) s;
+  GlobalConfig *cfg = log_pipe_get_config(s);
+#if 0
+  if (!filterx_expr_init(self->expr, cfg))
+    return FALSE;
+#endif
+
+  if (!self->name)
+    self->name = cfg_tree_get_rule_name(&cfg->tree, ENC_FILTER, s->expr_node);
+
+#if 0
```

Comment:
These blocks remained dead code by the end of this PR.

Reply:
done.